client/scss/components/_header.scss

```diff
@@ -0,0 +1,289 @@
+//* @define header; weak
+
+.header {
+  padding-top: 0.8rem;
+}
+
+.header__upper-wrap {
+  border-bottom: 1px solid color('keyline-grey');
+}
+
+.header__upper,
+.header__lower {
+  display: flex;
+}
+
+.header__upper {
+  padding-bottom: 0.8rem;
+  justify-content: flex-end;
+  position: relative;
+
+  @include respond-to('medium') {
+    justify-content: space-between;
+  }
+}
+
+.header__lower {
+  justify-content: space-between;
+  position: relative;
+
+  &.is-active {
```

Comment:
>  generic state hooks to any component

Then they shouldn't be nested and have their own component.
If, on a component per component level, the class does different things, it should probably be a component state, i.e. `--state`.

Either approach I think works, it just seems what we have now is a bit of a hybrid, which could cause confusion later on.